profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -231,83 +333,59 @@ export function DangerousIdentifier() {
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    const person = new FormData(e.target);
-    setPeople((ppl) => [...ppl, ...person]);
+    const data = new FormData(e.currentTarget);
+    const person = data.get("name");
+    setPeople((ppl) => [...ppl, { id: people.length, name: person }]);
   };
 
   return (
     <div>
       <form onSubmit={handleSubmit}>
-        <input type="text" />
+        <input type="text" name="name" />
         <button>Add Person</button>
       </form>
       <ul>
         {people.map((person) => (
-          <span key={person.name}>{person.name}</span>
+          <li key={person.id}>{person.name}</li>
         ))}
       </ul>
     </div>
   );
 }
 
-async function fetchLeader() {
-  return { name: "Messi" };
-}
-async function fetchDetails(leader) {
-  return { ...leader, country: "Argentina" };
-}
-
-// Hint: this only requires a single line change!
 export function UnnecessaryEffectTriggering() {
   const [leader, setLeader] = useState({});
 
   useEffect(() => {
-    const interval = setInterval(async () => {
-      const leader = await fetchLeader();
+    async function getData() {
+      const leader = await API.fetchLeader().then((l) => API.fetchDetails(l));
       setLeader(leader);
-    }, 1000);
-    return () => clearInterval(interval);
-  }, []);
-
-  useEffect(() => {
-    async function enhanceRecord() {
-      const enriched = await fetchDetails(leader);
-      setLeader(enriched);
     }
-    enhanceRecord();
-  }, [leader]);
+
+    getData();
+  }, []);
 
   return (
     <div>
-      <div>Leader:{leader.name}</div>
-      {leader.country && <div>{`From: ${leader.country}`}</div>}
+      <div>Leader: {leader.name}</div>
+      {leader.country && <div>From: {leader.country}</div>}
     </div>
   );
 }
 
-async function trackClick(ids) {
-  return ids;
-}
-
-// Hint: same error pattern as above
-export function IncorrectDependencies(records) {
-  const handleClick = useCallback(() => {
-    trackClick(records);
-  }, [records]);
```

Comment:
an empty dependency list could cause a stale record tracking, so proper way would be to either remove the useCallback hook or leave as is, but it doesn't serve much purpose in this component, it would if there were other pieces of state causing rerenders